llvm/lib/Transforms/Utils/Evaluator.cpp

```diff
@@ -145,9 +160,18 @@ static bool isSimpleEnoughPointerToCommit(Constant *C) {
         isa<GlobalVariable>(CE->getOperand(0)) &&
         cast<GEPOperator>(CE)->isInBounds()) {
       GlobalVariable *GV = cast<GlobalVariable>(CE->getOperand(0));
+      bool Pruning;
+      GlobalValue::LinkageTypes InitialLinkage;
+      std::tie(Pruning, InitialLinkage) = getPruningAndLinkage(*GV);
       // Do not allow weak/*_odr/linkonce/dllimport/dllexport linkage or
       // external globals.
-      if (!GV->hasUniqueInitializer())
+      if (!Pruning && !GV->hasUniqueInitializer())
```

Comment:
We test `Pruning` twice. Could this be:
````
if (Pruning) {
    if (...) {
        return false;
    }
} else {
   if (...) {
       return false; 
}
````
That would simplify these conditions a tiny bit.

Reply:
Thanks, I have updated the patch as your advice.